src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx

```diff
@@ -132,23 +132,25 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
                   variant="outlined"
                   error={!!errors.manufacturerName}
                   helperText={errors.manufacturerName?.message}
+                  onChange={(event) => {
+                    const selectedValue = event.target.value;
+                    if (selectedValue === 'createManufacturer') {
+                      const manufacturerName = prompt('Enter Manufacturer Name');
+                      if (manufacturerName) {
+                        createManufacturer(manufacturerName);
+                      } else {
+                        event.target.value = field.value;
+                      }
+                    }
+                    field.onChange(event);
```

Comment:
I would do this. I see why your way works, but this is a bit cleaner/ easier to tell what's going on :
```
                    if (selectedValue === 'createManufacturer') {
                      const manufacturerName = prompt('Enter Manufacturer Name');
                      if (manufacturerName) {
                        createManufacturer(manufacturerName);
                        field.onChange(event);
                      }
                    } else {
                      field.onChange(event);
                    }
```

Reply:
Thank you! I just made the changes.